src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HealthHeartbeatProvider.cs

```diff
@@ -0,0 +1,226 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+
+    /// <summary>
+    /// Implementation of health heartbeat functionality.
+    /// </summary>
+    internal class HealthHeartbeatProvider : IDisposable, IHeartbeatProvider
+    {
+        /// <summary>
+        /// The default interval between heartbeats if not specified by the user
+        /// </summary>
+        public static int DefaultHeartbeatIntervalMs = 5000;
```

Comment:
Does it need to be public? Can it be a `private const int`?

Reply:
I wanted to expos this for unit testing. Internal classes with public static members shouldn't be a big issue with the overall design of the SDK, although they may add noise for those working within the SDK... for testing sake I chose to keep it public. Perhaps a constants 'class' would be more appropriate?

Reply:
OK. Leave it the way it is, but maybe add to the doc comment something like: `Public so it can be updated by unit tests.`

Reply:
as long as class is internal it doesn't matter